style-guide/statements.md

```diff
@@ -0,0 +1,148 @@
+### Let statements
+
+There should be spaces after the `:` and on both sides of the `=` (if they are
+present). No space before the semi-colon.
+
+```rust
+// A comment.
+let pattern: Type = expr;
+
+let pattern;
+let pattern: Type;
+let pattern = expr;
+```
+
+If possible the declaration should be formatted on a single line. If this is not
+possible, then try splitting after the `=`, if the declaration can fit on two
+lines. The expression should be block indented.
+
+```rust
+let pattern: Type =
+    expr;
```

Comment:
I find this style hard to read; it obscures structure for me; I'd personally write this as (if it is long...):

```rust
let pattern
  : Type
  = expr;
```

Reply:
I think the guideline as currently stated is more consistent. There's a stronger cohesion between `pattern` and `Type` from there being no space before the colon.
That motivates having `pattern: Type` on the same line.

Reply:
@repax My point was mostly about `pattern` and `=` ; moving and aligning the `=` with the end of `let` becomes easier to scan imo; but I suppose it might not fit with the overall style.